pkg/command/root.go

```diff
@@ -125,9 +127,12 @@ func newRootCmd() *cobra.Command {
 			//   - 'config set' (would be a chicken and egg issue if user tries to set CEIP configuration
 			//      using "tanzu config set env.TANZU_CLI_CEIP_OPT_IN_PROMPT_ANSWER yes")
 			//   - 'config eula xxx', 'ceip set' (auto prompting when running these commands is confusing)
+			//   - 'tanzu pinniped-auth' ( since this command is being invoked by kubectl exec plugin, user doesn't
+			//      get to see the prompts and kubectl command execution just gets stuck and it is very hard for users to figure out)
 			if cmd.Name() != cobra.ShellCompRequestCmd && cmd.Name() != completionCmd.Name() &&
 				cmd.Name() != newVersionCmd().Name() &&
-				!(cmd.Name() == cmdNameSet && cmd.Parent().Name() == configCmd.Name()) {
+				!(cmd.Name() == cmdNameSet && cmd.Parent().Name() == configCmd.Name()) &&
+				!(cmd.Name() == cmdNamePinnipedAuth && cmd.Parent().Name() == cmdNameTanzu) {
 				if !(cmd.Parent().Name() == newEULACmd().Name()) &&
 					!(cmd.Name() == cmdNameSet && cmd.Parent().Name() == newCEIPParticipationCmd().Name()) {
 					if err := cliconfig.ConfigureEULA(false); err != nil {
```

Comment:
I like the generic approach to this code, using constants and command names.  However, it has grown to a point that I personally find it difficult to understand.  Furthermore, with this PR, three constants used here are simply defined further above, which does not really help make the code generic.

Are we at a point where it would be better to hard-code the commands we want to ignore?  We can use cobra's `cmd.CommandPath()` to help.  We could do something like this:
```
	// Prompt user for EULA and CEIP agreement if necessary, except for
	skipCommands := []string{
		// The shell completion setup is not interactive, so it should not trigger a prompt
		"tanzu __complete",
		"tanzu completion",
		// Common first command to run,
		"tanzu version",
		// It would be a chicken and egg issue if user tries to set CEIP configuration
		// using "tanzu config set env.TANZU_CLI_CEIP_OPT_IN_PROMPT_ANSWER yes"
		"tanzu config set",
		// Auto prompting when running these commands is confusing
		"tanzu config eula",
		"tanzu ceip-participation set",
		// This command is being invoked by the kubectl exec binary where the user doesn't
		// get to see the prompts and the kubectl command execution just gets stuck and it
		// is very hard for users to figure out what is going wrong
		"tanzu pinniped-auth",
	}
	skipPrompts := false
	for _, cmdPath := range skipCommands {
		if strings.HasPrefix(cmd.CommandPath(), cmdPath) {
			skipPrompts = true
			break
		}
	}
	if !skipPrompts {
            do the prompts
        }
       return nil
```
Do you think this increased readability is worth the reduced future-proof aspect?
@prkalle @vuil @anujc25

Reply:
Thanks @marckhouzam. I like this approach and it improves the readability. I will update the PR.